combustion-android-ble/src/main/java/inc/combustion/framework/ble/device/NodeBleDevice.kt

```diff
@@ -77,31 +78,37 @@ internal class NodeBleDevice(
         }
     }
 
-    private fun observeUartResponses(callback: (suspend (responses: List<NodeUARTMessage>) -> Unit)? = null) {
+    private fun observeUartMessages(callback: (suspend (messages: List<NodeUARTMessage>) -> Unit)? = null) {
         uart.jobManager.addJob(
             key = uart.serialNumber,
             job = uart.owner.lifecycleScope.launch(
                 CoroutineName("UartResponseObserver"),
             ) {
                 uart.observeUartCharacteristic {data ->
-                    val responses = NodeUARTMessage.fromData(data.toUByteArray())
-                    callback?.invoke(responses)
+                    val messages = NodeUARTMessage.fromData(data.toUByteArray())
+                    callback?.invoke(messages)
                 }
             }
         )
     }
-    private fun processUartResponses() {
-        observeUartResponses { responses ->
-            responses.forEach { response ->
-                when (response) {
+    private fun processUartMessages() {
+        observeUartMessages { messages ->
+            messages.forEach { message ->
+                when (message) {
                     is NodeReadFeatureFlagsResponse -> {
-                        readFeatureFlagsRequest.handled(response.success, response, response.requestId)
+                        readFeatureFlagsRequest.handled(message.success, message, message.requestId)
                     }
                     is GenericNodeResponse -> {
-                        genericRequestHandler.handled(response.success, response, response.requestId)
-                    } else -> {
+                        genericRequestHandler.handled(message.success, message, message.requestId)
+                    }
+                    is GenericNodeRequest -> {
+                        // Publish the request to the flow so it can be handled by the user.
+                        message.nodeSerialNumber = deviceInfoSerialNumber?: ""
+                        NetworkManager.flowHolder.mutableGenericNodeRequestFlow.emit(message)
+                    }
+                    else -> {
                         // drop the message
-                        Log.w(LOG_TAG, "NodeBLEDevice: Unhandled response: $response")
+                        //Log.w(LOG_TAG, "NodeBLEDevice: Unhandled response: $response")
```

Comment:
This warning still seems worthwhile.

Reply:
It spams the logs like crazy so I commented it out. This is receiving the same set of messages that are handled in the RepeatedProbleBleDevice. Maybe there is another way to filter those out? I'm not sure if that is possible since these are using the same flow.